cli/cmd/model_pull.go

```diff
@@ -0,0 +1,151 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/replicated/pkg/credentials"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+func (r *runners) InitModelPull(parent *cobra.Command) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "pull [NAME:TAG] [DESTINATION]",
+		Short:        "pull a model from the model repository",
+		Long:         `pull a model from the model repository`,
+		RunE:         r.pullModel,
+		Args:         cobra.ExactArgs(2),
+		SilenceUsage: true,
+	}
+	parent.AddCommand(cmd)
+
+	return cmd
+}
+
+func (r *runners) pullModel(cmd *cobra.Command, args []string) error {
+	// get the endpoint
+	endpoint, err := r.kotsAPI.GetModelsEndpoint()
+	if err != nil {
+		return err
+	}
+
+	if endpoint == "" {
+		return errors.New("Could not find models endpoint")
+	}
+	fmt.Printf("Pulling model from %s\n", endpoint)
```

Comment:
Is there a progress display when oras.Copy is called?

Reply:
not at this time